File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
        public int globalcount = 0;
public int result = 0;
public int kthSmallest(TreeNode root, int k) {
if(root == null) return 0;
kthSmallest(root.left, k);
globalcount++;
if(globalcount == k) result = root.val;
kthSmallest(root.right, k);
return result;
}
    
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    int ans=0;
    int count=0;
    int kthSmallest(TreeNode* root, int k) {
        
        if(root==NULL)return 0;
        
        kthSmallest(root->left,k);
        {
            count=count+1;
            if(count==k)ans=root->val;
            
        }
        kthSmallest(root->right,k);
        
        return ans;
    }
};